include/Event.hpp

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <cstdint>
+#include <functional>
+
+class Event {
+public:
+    Event(std::uint64_t time, const std::function<void()>& handler);
+
+    void Execute() const;
+
+    std::uint64_t m_execution_time;
```

Comment:
Nit: `m_` for private members only to distinguish from public data (https://lefticus.gitbooks.io/cpp-best-practices/content/03-Style.html).